frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -3455,6 +3458,190 @@ def reinterpret(self, x: Conversion, *, bit_width: int) -> Conversion:
         )
         return self.operation(operation, resulting_type, x.result)
 
+    def where(
+        self,
+        resulting_type: ConversionType,
+        condition: Conversion,
+        when_true: Conversion,
+        when_false: Conversion,
+    ) -> Conversion:
+        if condition.original_bit_width != 1 or condition.is_signed:
+            highlights = {
+                condition.origin: "condition is not uint1",
+                self.converting: "but it needs to be for where operation",
+            }
+            self.error(highlights)
+
+        if when_true.is_clear:
+            highlights = {
+                when_true.origin: "outcome of true condition is not encrypted",
+                self.converting: "but it needs to be for where operation",
+            }
+            self.error(highlights)
+
+        if when_false.is_clear:
+            highlights = {
+                when_false.origin: "outcome of false condition is not encrypted",
+                self.converting: "but it needs to be for where operation",
+            }
+            self.error(highlights)
+
+        chunk_size = self.configuration.multivariate_where_chunk_size
+        chunk_type = self.eint(chunk_size + 1)
+
+        if condition.bit_width != 1:
+            shifter = self.constant(self.i(condition.bit_width + 1), 2 ** (condition.bit_width - 1))
+            condition = self.mul(condition.type, condition, shifter)
+
+        condition = self.reinterpret(condition, bit_width=chunk_type.bit_width)
+
+        when_true_chunks = []
+        when_true_intermediate_type = self.tensor(
+            self.eint(resulting_type.bit_width),
+            shape=(np.zeros(condition.shape) + np.zeros(when_true.shape)).shape,
+        )
+
+        cursor = 0
```

Comment:
can't it be a for loop on branch side ? the code is completely the same apart some padding with zeros (shifting TLU).
also would be simpler to keep the same TLU and reverse the condition for the second TLU.
1. simplify code, 2. enable full tensorized version (bit extract + TLU).

Reply:
I'll introduce a helper function to abstract it away.
As for tensorization, like with ReLU, let's leave it out of this PR for the moment. It's big enough already, it can be done in another PR.

Reply:
I recall we spoke earlier of a general retensorization so we don't need to do it on each case.